foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/platform/database/DB2ZPlatform.java

```diff
@@ -475,4 +475,47 @@ protected void setNullFromDatabaseField(DatabaseField databaseField, CallableSta
             super.setNullFromDatabaseField(databaseField, statement, name);
         }
     }
+
+    @Override
+    public Object getParameterValueFromDatabaseCall(CallableStatement statement, String name, AbstractSession session)
+                throws SQLException {
+        String methodName = null;
+        Class[] methodArgs = null;
+        Object[] parameters = null;
+
+        methodName = "getJccObjectAtName";
+        methodArgs = new Class[] {String.class};
+        parameters = new Object[] {name};
+
+        if(methodName != null) {
+            try {
+                Class clazz = null;
+                Method method = null;
+                if (PrivilegedAccessHelper.shouldUsePrivilegedAccess()) {
+                    try {
+                        ClassLoader cl = AccessController.doPrivileged(new PrivilegedGetContextClassLoader(Thread.currentThread()));
+                        clazz = AccessController.doPrivileged(new PrivilegedClassForName(DB2_CALLABLESTATEMENT_CLASS, true, cl));
+                        method = AccessController.doPrivileged(new PrivilegedGetMethod(clazz, methodName, methodArgs, true));
+                        Object o = statement.unwrap(clazz);
+                        return AccessController.doPrivileged(new PrivilegedMethodInvoker(method, o, parameters));
+                    } catch (PrivilegedActionException ex) {
+                        if (ex.getCause() instanceof ClassNotFoundException) {
+                            throw (ClassNotFoundException) ex.getCause();
+                        }
+                        throw (RuntimeException) ex.getCause();
+                    }
+                } else {
+                    ClassLoader cl = PrivilegedAccessHelper.getContextClassLoader(Thread.currentThread());
+                    clazz = PrivilegedAccessHelper.getClassForName(DB2_CALLABLESTATEMENT_CLASS, true, cl);
+                    method = PrivilegedAccessHelper.getMethod(clazz, methodName, methodArgs, true);
+                    Object o = statement.unwrap(clazz);
+                    return PrivilegedAccessHelper.invokeMethod(method, o, parameters);
+                }
+            } catch (ClassNotFoundException | NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
```

Comment:
Have you considered catching just "ReflectiveOperationException" instead of listing almost all its subclasses?

Reply:
Fair enough. I hadn't actually looked into a common Exception type; I just had Eclipse build the catch and went with that. Thanks!